File: Java/Count Number of Distinct Integers After Reverse Operations.java
import java.util.HashSet;

public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(n)
     */
    public int countDistinctIntegers(int[] nums) {
        HashSet<Integer> uniq = new HashSet<>();

        for (int num : nums) {
            uniq.add(num);
            int reverse = 0;

            for (; num != 0; num /= 10)
                reverse = reverse * 10 + num % 10;

            uniq.add(reverse);
        }

        return uniq.size();
    }
}

File: Python/Count Number of Distinct Integers After Reverse Operations.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(n)
    """

    def countDistinctIntegers(self, nums: List[int]) -> int:
        uniq = set()

        for num in nums:
            uniq.add(num)
            uniq.add(int(str(num)[::-1]))

        return len(uniq)
